### ISSUE TITLE
Reproducible issue

### ISSUE THREAD

**Issue by bywbilly** [mileyan/DARENet#3]
I recently try to reproduce your result on Market1501 dataset with both denenet and resnet, however, I find errors below:

My running command is:
`./train.sh dare_D 0 path_to_train_folder None Market1501`

And here is the error:
![image](https://user-images.githubusercontent.com/10766902/49045300-957e2880-f19e-11e8-9ce2-4160109aaa21.png)

**Comment by YurongYou** [comment]
Use pytorch 0.2.0

**Comment by ChrystleMyrnaLobo** [comment]
You need to use BatchNorm1d (instead of BatchNorm2d) for fully connected layers. For example in the `ResNet` class in models/dare_resnet.py 
```
self.layer1_fc = nn.Sequential(
            nn.Linear(64 * block.expansion, fc_layer1),
            # nn.BatchNorm2d(fc_layer1),
            nn.BatchNorm1d(fc_layer1),
            nn.ReLU(inplace=True),
            nn.Dropout(p=drop_rate),
            nn.Linear(fc_layer1, fc_layer2),
        )
```
